doc/test/README.md

```diff
@@ -0,0 +1,2 @@
+# Azure SDK Test Best Practices
+
```

Comment:
Do we want to link to the other two docs here/give some top level overview?

Reply:
I actually removed this directory and file in favor of the `doc/dev/tests.md` file.